controllers/storagecluster/provider_server.go

```diff
@@ -189,6 +214,35 @@ func (o *ocsProviderServer) ensureDeploymentReplica(actual, desired *appsv1.Depl
 	return nil
 }
 
+// getWorkerNodesInternalIPAddresses return slice of Internal IPAddress of worker nodes
+func (r *StorageClusterReconciler) getWorkerNodesInternalIPAddresses() ([]string, error) {
+
+	nodes := &corev1.NodeList{}
+
+	err := r.Client.List(context.TODO(), nodes)
+	if err != nil {
+		r.Log.Error(err, "Failed to list nodes")
+		return nil, err
+	}
+
+	nodeAddresses := []string{}
+
+	for _, node := range nodes.Items {
+		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/worker"]; ok {
+			for _, address := range node.Status.Addresses {
```

Comment:
Not sure about the size of the address structure, but the last comment reasoning might also apply here.

As a general note using `for _, item := range slice` should be used when the slide slice either holds small data types (e.g numbers, booleans, etc.) or pointers.

Reply:
This will have only 2 fields keeping it as it is